### PR TITLE
DeformablePSROIPoolig reference implementation

### PR DESCRIPTION
**Pull request by mitruska**
### Details:
- [x] Reference implementation
- [x] Additional output_dim value check in ng op
- [x] Backend tests
- [x] Single layer tests + serialize
- [x] Visit attributes test 
- [x] CPU implementation fix (Typo, X swapped with Y)

### Tickets:
 - 37509 (45724, 45725)

### CODE REVIEW

**Comment by iefode** [comment]
Please add the op name to [this list](https://github.com/openvinotoolkit/openvino/blob/a478475386732c3332c701d210786af7dad96eed/inference-engine/tests/ie_test_utils/functional_test_utils/layer_tests_summary/summarize.py#L24)

**Review by pelszkow — APPROVED** [review]

File: ngraph/test/runtime/interpreter/evaluates_map.cpp [hunk 1]
```diff
@@ -2126,6 +2127,48 @@ namespace
 
         return true;
     }
+  template <element::Type_t ET>
+    bool evaluate(const shared_ptr<op::v1::DeformablePSROIPooling>& op,
+                  const HostTensorVector& outputs,
+                  const HostTensorVector& inputs)
+    {
+        using T = typename element_type_traits<ET>::value_type;
+        if (inputs.size() == 2)  // no offset input
```
[the remark this answers is not in the document]
Comment (mitruska) [reply ngraph/test/runtime/interpreter/evaluates_map.cpp]: Done

File: ngraph/core/reference/include/ngraph/runtime/reference/deformable_psroi_pooling.hpp [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+                    const float bin_width = roi_width / static_cast<float>(width_out);
+                    const float bin_height = roi_height / static_cast<float>(height_out);
+
+                    size_t c_idx_in = 0;
+                    for (size_t c_idx_out = 0; c_idx_out < channels_out; ++c_idx_out)
+                    {
+                        for (size_t h_idx_out = 0; h_idx_out < height_out; ++h_idx_out)
+                        {
+                            for (size_t w_idx_out = 0; w_idx_out < width_out; ++w_idx_out)
```
[the remark this answers is not in the document]
Comment (mitruska) [reply ngraph/core/reference/include/ngraph/runtime/reference/deformable_psroi_pooling.hpp]: Done

**Review by ggalieroc — COMMENTED** [review]
Please add Serialization SLT tests

File: ngraph/core/reference/include/ngraph/runtime/reference/deformable_psroi_pooling.hpp [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            namespace details
+            {
+                template <typename T>
+                T clamp(const T& value, const T& lower, const T& upper)
```
Comment (ilyachur) [inline ngraph/core/reference/include/ngraph/runtime/reference/deformable_psroi_pooling.hpp:23]: Maybe can we reuse clamp reference implementation?

Comment (mitruska) [reply ngraph/core/reference/include/ngraph/runtime/reference/deformable_psroi_pooling.hpp]: It seems we can, done:)

**Review by a-sidorova — APPROVED** [review]

**Comment by ilyachur** [comment]
@mitruska Please resolve merge conflicts

**Comment by mitruska** [comment]
Let's continue the PR review process and wait with merging for a while.

**Comment by jdanieck** [comment]
TPP file updates regarding needs of DeformablePSROIPooling are in #5607.

**Comment by jdanieck** [comment]
@ilyachur I guess we need only your review here. Besides [updates to TPP](https://github.com/openvinotoolkit/openvino/pull/5607), is there anything else which are blocking us ?

**Comment by jdanieck** [comment]
> > @ilyachur I guess we need only your review here. Besides [updates to TPP](https://github.com/openvinotoolkit/openvino/pull/5607), is there anything else which are blocking us ?
> 
> I will have a look. At least before the merge all conflicts should be resolved

As Kasia is on holidays I'll try to resolve conflicts. I should have permissions to push to this branch.

**Comment by ilyachur** [comment]
> > > @ilyachur I guess we need only your review here. Besides [updates to TPP](https://github.com/openvinotoolkit/openvino/pull/5607), is there anything else which are blocking us ?
> > 
> > 
> > I will have a look. At least before the merge all conflicts should be resolved
> 
> As Kasia is on holidays I'll try to resolve conflicts. I should have permissions to push to this branch.

Good, it is strange that conflicts can be resolved from the web interface.

**Comment by jdanieck** [comment]
> > > > @ilyachur I guess we need only your review here. Besides [updates to TPP](https://github.com/openvinotoolkit/openvino/pull/5607), is there anything else which are blocking us ?
> > > 
> > > 
> > > I will have a look. At least before the merge all conflicts should be resolved
> > 
> > 
> > As Kasia is on holidays I'll try to resolve conflicts. I should have permissions to push to this branch.
> 
> Good, it is strange that conflicts can be resolved from the web interface.

**Comment by jdanieck** [comment]
@ttroilov please verify that [updated license header](https://github.com/openvinotoolkit/openvino/pull/5116/commits/6f2750c8508701a166d98b5d09931dcdcf5a2a3c) is correct.

**Review by ttroilov — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Pull request merged**